CBinding/CMake/CMakeFileFormat.cs

```diff
@@ -0,0 +1,289 @@
+				foreach (Match m in Regex.Matches (contentFile.Text, fullPattern, RegexOptions.Singleline)) {
+					if (m.Value.StartsWith ("#"))
+						continue; //Skip comments.
+					yield return m;
+				}
+            }
+		}
+
+		void parseCommand (Match commandMatch)
+		{
+			string commandName = Regex.Match (commandMatch.Value, @"^[A-Za-z_][A-Za-z0-9_]*").Value;
+			CMakeCommand command = new CMakeCommand (commandName, commandMatch, this);
+
+			switch (command.Name.ToLower ()) {
+			case "if":
+				blocks.Push ("if");
+				break;
+			case "elseif":
+				blocks.Pop ();
+				blocks.Push ("elseif");
+				break;
+			case "else":
+				blocks.Pop ();
+				blocks.Push ("else");
+				break;
+			case "endif":
+				blocks.Pop ();
+				break;
+			case "while":
+				blocks.Push ("while");
+				break;
+			case "endwhile":
+				blocks.Pop ();
+				break;
+			case "foreach":
+				blocks.Push ("foreach");
+				break;
+			case "project":
+				ProjectName = command.Arguments[0].ToString();
+				break;
+			case "set":
+				SetCommands.Add (string.Format ("{0}:{1}", command.Arguments [0].GetValues () [0], commandMatch.Index), command);
+				break;
+			case "add_library":
+			case "add_executable":
+				CMakeTarget target = new CMakeTarget (command, this);
+
+				if (!target.IsAliasOrImported) {
+					Targets.Add (string.Format ("{0}:{1}", target.Name, commandMatch.Index), target);
+					target.PrintTarget ();
+				}
+				break;
+			case "add_subdirectory":
+				if (Parent == null) {
+					FilePath temp = new FilePath (command.Arguments [0].ToString ());
+					if (!Children.ContainsKey (temp))
+						Children.Add (temp, new CMakeFileFormat (temp, Project, this));
+				} else {
+					FilePath temp = new FilePath (command.Arguments [0].ToString ());
+					if (!Parent.Children.ContainsKey (temp))
+						Children.Add (temp, new CMakeFileFormat (temp, Project, Parent));
+				}
+				break;
+			default:
+				break;
+			}
+			
+			command.IsEditable = blocks.Count == 0;
+			
+			allCommands.Add (string.Format ("{0}:{1}", commandName, commandMatch.Index), command);
+		}
+
+		public void Parse ()
+		{
+			Targets.Clear ();
+			SetCommands.Clear ();
+			allCommands.Clear ();
+			blocks.Clear ();
+			Children.Clear ();
+			ProjectName = "";
+			
+			contentFile = new MonoDevelop.Projects.Text.TextFile(File);
+			foreach (Match m in readNextCommand ()) {
+				parseCommand (m);
+			}
+			
+			if (blocks.Count > 0)
+				throw new Exception ("Unmatched block command (if, while or foreach).");
+		}
+
+		//for debugging purposes. 
+		void printArgs ()
+		{
+			foreach (var command in SetCommands)
+			{
+				LoggingService.LogDebug (String.Format ("{0}\n{1}", command.Key, command.Value.ToString ()));
+			}
+		}
+		
+		public void NewTarget (string name, CMakeTarget.Types type)
+		{
+			string commandName = "";
+			string libraryType = "";
+			
+			switch (type) {
+			case CMakeTarget.Types.BINARY:
```

Comment:
C# convention would usually be to use PascalCase, e.g. CMakeTarget.Types.Binary and CMakeTarget.Types.SharedLibrary.